mobile/envoy_build_config/extension_registry.cc

```diff
@@ -74,64 +74,123 @@ namespace Envoy {
-  Extensions::StatSinks::Statsd::forceRegisterStatsdSinkFactory();
-  Extensions::TransportSockets::Http11Connect::
-      forceRegisterUpstreamHttp11ConnectSocketConfigFactory();
+  // TODO(alyssawilk) move to the listener build.
   Extensions::TransportSockets::RawBuffer::forceRegisterDownstreamRawBufferSocketFactory();
-  Extensions::TransportSockets::RawBuffer::forceRegisterUpstreamRawBufferSocketFactory();
+  // This is the default certificate validator, still compiled by default but hopefully soon to be
+  // deprecated in production by iOS and Android platform validators.
   Extensions::TransportSockets::Tls::forceRegisterDefaultCertValidatorFactory();
+  // This is the base for the still-being-validated platform validators.
   Extensions::TransportSockets::Tls::forceRegisterPlatformBridgeCertValidatorFactory();
+
+  // This transport socket handles upstream TLS connections.
   Extensions::TransportSockets::Tls::forceRegisterUpstreamSslSocketFactory();
+  // This transport socket handles doing CONNECT requests to any configured system proxies.
+  Extensions::TransportSockets::Http11Connect::
+      forceRegisterUpstreamHttp11ConnectSocketConfigFactory();
   Extensions::Upstreams::Http::forceRegisterProtocolOptionsConfigFactory();
+  // This transport socket handles plaintext (http) traffic.
+  Extensions::TransportSockets::RawBuffer::forceRegisterUpstreamRawBufferSocketFactory();
+
+  // This is the default HTTP connection pool factory required for L7 upstream traffic.
   Extensions::Upstreams::Http::Generic::forceRegisterGenericGenericConnPoolFactory();
+  // This is the default TCP client connection factory required for L7 upstream traffic.
+  Network::forceRegisterDefaultClientConnectionFactory();
+  // This is the default socket factory required for L7 traffic.
+  Network::forceRegisterSocketInterfaceImpl();
+  // This is the RE factory, required at least if any config uses regex, which stats do.
+  Regex::forceRegisterGoogleReEngineFactory();
+
+  // These are required to support specific route configs, if they are on.
+  // It's likely no current users of E-M require them so we could optionally compile out by default.
+  Router::forceRegisterDefaultEarlyDataPolicyFactory();
+  Router::forceRegisterRouteListMatchActionFactory();
+  Router::forceRegisterRouteMatchActionFactory();
   Extensions::UriTemplate::Match::forceRegisterUriTemplateMatcherFactory();
   Extensions::UriTemplate::Rewrite::forceRegisterUriTemplateRewriterFactory();
   Http::Matching::forceRegisterHttpRequestHeadersDataInputFactory();
   Http::Matching::forceRegisterHttpRequestTrailersDataInputFactory();
   Http::Matching::forceRegisterHttpResponseHeadersDataInputFactory();
   Http::Matching::forceRegisterHttpResponseTrailersDataInputFactory();
-  Network::Address::forceRegisterIpResolver();
-  Network::forceRegisterDefaultClientConnectionFactory();
+
+  // Envoy Mobile uses the GetAddrInfo resolver for DNS lookups by default.
```

Comment:
May be worth calling out that it's used only on Android and that we could probably compile it out on iOS